src/web/mjs/connectors/WanPaMan.mjs

```diff
@@ -0,0 +1,61 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class OnePunchMan extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'wanpaman';
+        super.label = 'Wan-Pa Man';
+        this.tags = [ 'manga', 'japanese'];
+        this.url = 'http://galaxyheavyblow.web.fc2.com';
+    }
+
+    async _getMangas() {
+        return [
+            {
+                id: this.url,
+                title: this.label
+            }
+        ];
+    }
+
+    async _getChapters(manga) {
+        let request = new Request(new URL(manga.id), this.requestOptions);
+        let script = `new Promise(resolve => {
+            let chapters = [...document.querySelectorAll('body a[href*="imageviewer"]')].map(link => {
+                return {
+                    id: link.href,
+                    title: link.text.trim()
+                };
+            });
+            resolve(chapters.reverse());
+        });`;
+
+        return Engine.Request.fetchUI( request, script );
+    }
+
+    async _getPages(chapter) {
+        let aid = chapter.id.match(/(?:aid=)(\d+)/)[1];
```

Comment:
May use URL object to extract certain data from an URL, instead of custom regex,
e.g.
```javascript
let uri= new URL(chapter.id, this.url);
let aid = uri.searchParams.get('aid');
let iid = uri.searchParams.get('iid');
let path = uri.pathname;
// ...
let rest = [path, '', aid, iid, ''].join('/');
```

Reply:
Ohh nice, I really like that.